### PR TITLE
feat: support gecko bootloader firmwares

### LINKED ISSUE #1173: Support GBL format for OTA firmware upgrades of Series 700 devices
In order to perform firmware upgrades on SiLabs Series 700 devices,  we need to support the Gecko Bootloader (GBL) file format.

**Describe the solution you'd like**
Add the GBL format to the  guessFormat function in zwavejs.

**Describe alternatives you've considered**

**Additional context**

* Sample GBL firmware files: Update Recessed Door Sensor 7 V1.05 <https://aeotec.freshdesk.com/support/solutions/articles/6000243778-update-recessed-door-sensor-7-v1-05>)"
* Gecko Bootloader User Guide  https://www.wless.ru/files/Silabs/EFR32/ug266-gecko-bootloader-user-guide.pdf

### PR DESCRIPTION
**Pull request by AlCalzone**
fixes: #1173 

@larstobi you might wanna test this :)

### CODE REVIEW

**Comment by larstobi** [comment]
@AlCalzone Wow, that was fast! All right, I'll test it, but will have to buy a Series 700 device first. I'll get right on that!

**Comment by AlCalzone** [comment]
LOL I thought you already had one :D

**Comment by larstobi** [comment]
No, haha, but I will soon. I have a "Aeotec Door/ Window Sensor 7" on the way to my doorstep!

File: packages/core/src/util/firmware.ts [hunk 1]
```diff
@@ -2,13 +2,20 @@
 import MemoryMap from "nrf-intel-hex";
 import { ZWaveError, ZWaveErrorCodes } from "../error/ZWaveError";
 
-export type FirmwareFileFormat = "aeotec" | "otz" | "ota" | "hex";
+export type FirmwareFileFormat = "aeotec" | "otz" | "ota" | "hex" | "gecko";
```
Comment (robertsLando) [inline packages/core/src/util/firmware.ts:5]: @AlCalzone What about making this an enum? So we prevent to have hardcoded strings

File: packages/core/src/util/firmware.ts [hunk 2]
```diff
@@ -21,7 +28,7 @@ export function guessFirmwareFileFormat(
 ): FirmwareFileFormat {
 	if (
 		(filename.endsWith(".exe") || filename.endsWith(".ex_")) &&
-		rawData.includes(Buffer.from("Aeon Labs", "utf8"))
+		rawData.includes(firmwareIndicators.aeotec)
 	) {
 		return "aeotec";
```
Comment (robertsLando) [inline packages/core/src/util/firmware.ts:33]: This could become `FirmwareFileFormat.aeotec`

File: packages/core/src/util/firmware.ts [hunk 3]
```diff
@@ -31,6 +38,11 @@ export function guessFirmwareFileFormat(
 			"Encrypted .hec firmware files are not supported",
 			ZWaveErrorCodes.Unsupported_Firmware_Format,
 		);
+	} else if (
+		filename.endsWith(".gbl") &&
+		rawData.readUInt32BE(0) === firmwareIndicators.gecko
+	) {
+		return "gecko";
```
Comment (robertsLando) [inline packages/core/src/util/firmware.ts:45]: `FirmwareFileFormat.gecko`

File: packages/core/src/util/firmware.ts [hunk 4]
```diff
@@ -58,6 +71,10 @@ export function extractFirmware(
 		case "ota":
 		case "hex":
 			return extractFirmwareHEX(rawData);
+		case "gecko":
```
Comment (robertsLando) [inline packages/core/src/util/firmware.ts:74]: same

**Review by robertsLando — CHANGES_REQUESTED** [review]

File: packages/core/src/util/firmware.ts (hunk 1 above)
Comment (AlCalzone) [reply packages/core/src/util/firmware.ts:5]: I don't really see the benefit of this (breaking!) change when there is auto-completion and an exported type to help the discoverability.

Comment (robertsLando) [reply packages/core/src/util/firmware.ts:5]: Ok no problem :)

**Review by robertsLando — APPROVED** [review]
LGTM if @larstobi test completes correctly :)

**Comment by larstobi** [comment]
@AlCalzone @robertsLando 

Using the latest zj2m (zwave-js 5.7.0) docker image from docker hub, I've tested this on the Aeotec Recessed Door Sensor 7 with the EU gbl file from this article: https://aeotec.freshdesk.com/support/solutions/articles/6000243778-update-recessed-door-sensor-7-v1-05

"Error while calling api beginFirmwareUpdate: Unable to extract firmware from file: could not detect firmware format"

**Comment by AlCalzone** [comment]
:( let me check

**Comment by AlCalzone** [comment]
This seems to be a problem outside of zwave-js. This is the method that throws in your case:
grafik

@robertsLando Any idea?

**Comment by robertsLando** [comment]
@larstobi Could you show me the exact args that are sent to the api? There should be a log for them

**Comment by larstobi** [comment]
@robertsLando I have turned on debug logging to file, as well as web ui console debug logging, but I cannot find any args logged. Where should they appear?

**Comment by robertsLando** [comment]
You should see a log that says you have called an api request `beginFirmwareUpdate` and then the args. Are you using latest from master?

**Pull request merged**